rpmlint/checks/BinariesCheck.py

```diff
@@ -452,6 +454,24 @@ def _check_executable_shlib(self, pkg, pkgfile_path, path):
             if interp:
                 self.output.add_info('E', pkg, 'shared-library-not-executable', path)
 
+    def _check_optflags(self, pkg, pkgfile_path, path):
+        if self.readelf_parser.is_archive:
+            return
+
+        mandatory_optflags = self.config.configuration['MandatoryOptflags']
+        forbidden_optflags = self.config.configuration['ForbiddenOptflags']
+        if not mandatory_optflags and not forbidden_optflags:
+            return
+
+        for dwarf_unit in self.objdump_parser.compile_units:
+            tokens = dwarf_unit['producer'].split(' ')
+            missing = [mo for mo in mandatory_optflags if mo not in tokens]
+            forbidden = [f for f in forbidden_optflags if f in tokens]
+            if missing:
+                self.output.add_info('W', pkg, 'missing-mandatory-optflags', path, ' '.join(missing))
+            if forbidden:
+                self.output.add_info('W', pkg, 'forbidden-optflags', path, ' '.join(forbidden))
```

Comment:
This can be error for sure, if you set out to detect optflag and go you way to set to detect it you should explode if it is not allowed.